assets/wizards/engagement/components/prompt.js

```diff
@@ -0,0 +1,294 @@
+/* eslint-disable no-nested-ternary */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { BaseControl, CheckboxControl, TextareaControl } from '@wordpress/components';
+import apiFetch from '@wordpress/api-fetch';
+import { Fragment, useEffect, useState } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import { stringify } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ActionCard,
+	Button,
+	Grid,
+	ImageUpload,
+	Notice,
+	TextControl,
+	WebPreview,
+} from '../../../components/src';
+
+export default function Prompt( { inFlight, prompt, setInFlight, setPrompts } ) {
```

Comment:
Just now realized that this object is [this object](https://github.com/Automattic/newspack-popups/blob/428bb98636c4d97eb2312a3583108204543f6739/includes/schemas/class-prompts.php). Not sure if it still warrants the same thing. Maybe a link to the schema suffice?

Reply:
Ah yes, 46801d5 adds a comment with a link to the Prompt type schema in Newspack Campaigns. It does feel a little weird to have one component as TSX with schema defined there, and this component in JSX with the schema defined in PHP, but it would probably also be redundant to re-define the Prompt schema in TS here, right?

Reply:
I wouldn't say redundant if we see this as a generic component that can be used in any context. It would also be good to track the compatibility of the component with the actual schema. If the schema changes we'd see a type error, which is better than whatever error it may cause to the component.

Reply:
4cc1730 and e3c776a refactor prompt.js to TypeScript.